src/main/java/org/jenkinsci/plugins/workflow/cps/CpsBodyExecution.java

```diff
@@ -243,13 +239,12 @@ public Collection<StepExecution> getCurrentExecutions() {
     }
 
     @Override
-    public boolean cancel(final CauseOfInterruption... causes) {
+    public boolean cancel(Throwable error) {
         // 'stopped' and 'thread' are updated atomically
         CpsThread t;
         synchronized (this) {
             if (isDone())  return false;   // already complete
-            // TODO should perhaps rather override cancel(Throwable) and make this overload just delegate to that one
-            stopped = new FlowInterruptedException(Result.ABORTED, causes);
```

Comment:
With these changes, any code that currently calls `cancel(Throwable)` with an exception that is not a `FlowInterruptedException` will no longer result in a `FlowInterruptedException` at runtime, so behavior would change. Did you check the PCT? I am not sure how many callers of `cancel(Throwable)` exist.

Reply:
> I am not sure how many callers of `cancel(Throwable)` [that do not pass `FlowInterruptedException`] exist.

I could not find any in common Pipeline plugins. There seem to be a few plugins still improperly overriding `stop` (not updated like e.g. https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/49) that I could locate in my local Maven index:

* `DatabaseConnectionStep`
* `ExwsExecution`
* `GitLabBuildsStep`
* `GitLabCommitStatusStep`

> behavior would change

Indeed it would, though probably for the better—the precise reason for the failure of the body would be thrown up naturally.

Reply:
Ok, that seems fine. I was just worried that if there was widespread usage then this might cause a bunch of test failures because of things like build results changing from `ABORTED` to `FAILURE`.

Reply:
Checked `usage-in-plugins` with `-i -M` on

```
org.jenkinsci.plugins.workflow.steps.BodyExecution#cancel
```

resulting in

```
    "org/jenkinsci/plugins/workflow/steps/BodyExecution#cancel(Ljava/lang/Throwable;)Z": [
      "applitools-eyes",
      "browserstack-integration",
      "database",
      "external-workspace-manager",
      "gitlab-plugin",
      "performance-signature-dynatrace",
      "performance-signature-dynatracesaas",
      "pipeline-gitstatuswrapper",
      "pipeline-npm",
      "sauce-ondemand",
      "testingbot",
      "workflow-cps",
      "workflow-step-api"
    ]
```

so looks like the previously mentioned hits plus a few more obscure ones. Checking these quickly, they are all `StepExecution.stop` overrides. This idiom is deprecated but unlikely to lead to a behavioral change anyway, since if a block-scoped step is running another step in its body, `stop` will be called on the innermost.